File: backend/src/main/java/com/projetin/projetospringangular/model/Produto.java
package com.projetin.projetospringangular.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@Entity
@Table(name = "produtos")
public class Produto  {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;

    @NotNull
    @Enumerated(EnumType.STRING)
    private Categoria categoria;

    private String descricao;
    private BigDecimal preco;
    private Integer quantidadeEstoque;
    private String codigoBarras;
    private Boolean ativo = true;

}

File: frontend/src/app/models/produto.ts
export interface Produto {
  id?: number;              // Long -> number (o ? é porque no POST o ID não existe ainda)
  nome: string;             // String -> string
  categoria: string;        // O Enum no TS tratamos como string inicialmente para simplificar
  descricao: string;
  preco: number;            // BigDecimal -> number
  quantidadeEstoque: number; // Integer -> number
  codigoBarras?: string;    // String -> string (opcional conforme seu desafio)
  ativo: boolean;           // Boolean -> boolean
}

